src/test/java/core/basesyntax/strategy/StrategyTest.java

```diff
@@ -0,0 +1,98 @@
+package core.basesyntax.strategy;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Operation;
+import core.basesyntax.model.TransactionDto;
+import core.basesyntax.strategy.strategies.BalanceStrategy;
+import core.basesyntax.strategy.strategies.PurchaseStrategy;
+import core.basesyntax.strategy.strategies.ReturnStrategy;
+import core.basesyntax.strategy.strategies.SupplyStrategy;
+import org.junit.Test;
+
+public class StrategyTest {
+    private static final Fruit BANANA = new Fruit("banana");
+    private static final Fruit APPLE = new Fruit("apple");
+    private TransactionDto dto;
+
+    @Test(expected = Test.None.class)
+    public void validData() {
+        dto = new TransactionDto(Operation.BALANCE, BANANA, 50);
+        OperationStrategy.chooseTheStrategy(dto);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void invalidOperation() {
+        dto = new TransactionDto(Operation.valueOf("CATCH"), APPLE, 60);
+        OperationStrategy.chooseTheStrategy(dto);
+    }
+
+    @Test
+    public void balanceStrategyCheck_OK() {
+        dto = new TransactionDto(Operation.BALANCE, BANANA, 40);
+        new BalanceStrategy().apply(dto);
+        assertNotNull(Storage.storage);
```

Comment:
let's check the state of the storage after applying each operation - and don't forget to clear the storage